datetime/src/main/java/com/fasterxml/jackson/datatype/jsr310/deser/DurationDeserializer.java

```diff
@@ -79,20 +88,38 @@ protected DurationDeserializer withLeniency(Boolean leniency) {
                     deser = deser.withLeniency(leniency);
                 }
             }
+            if (format.hasPattern()) {
+                this.pattern = DurationPattern.from(format.getPattern())
+                        .orElse(null);
+            } else {
+                this.pattern = null;
+            }
         }
         return deser;
     }
 
+    private DurationDeserializer withPattern(DurationPattern pattern) {
+        this.pattern = pattern;
+        return this;
+    }
+
     @Override
     public Duration deserialize(JsonParser parser, DeserializationContext context) throws IOException
     {
         switch (parser.currentTokenId())
         {
             case JsonTokenId.ID_NUMBER_FLOAT:
-                BigDecimal value = parser.getDecimalValue();
-                return DecimalUtils.extractSecondsAndNanos(value, Duration::ofSeconds);
+                BigDecimal decValue = parser.getDecimalValue();
+                if (pattern != null) {
+                    return pattern.parse(decValue.longValue());
+                }
+                return DecimalUtils.extractSecondsAndNanos(decValue, Duration::ofSeconds);
             case JsonTokenId.ID_NUMBER_INT:
-                return _fromTimestamp(context, parser.getLongValue());
+                long intValue = parser.getLongValue();
+                if (pattern != null) {
+                    return pattern.parse(intValue);
+                }
+                return _fromTimestamp(context, intValue);
```

Comment:
So the new functionality will override the flag `DeserializationFeature.READ_DATE_TIMESTAMPS_AS_NANOSECONDS` if already set... guessing that is ok, though to be safer you might want to adhere to the flag if both are set, and there have a conflict, i.e they don't specify nanos in the pattern. The documentation should be updated too.

Reply:
I think not honouring the flag is ok since the `@JsonFormat` annotation has been used explicitly may have more precedence. Actually that flag is not well used in this serializer because it uses `ofSeconds` while the flag is about nanos.

Reply:
BTW, what docs should I update?

Reply:
I think a line here, to add Duration to the list of exceptions, and mention that it supports "pattern", would be helpful (in the section after the line below)

> Some exceptions to this standard serialization/deserialization rule:

https://github.com/FasterXML/jackson-modules-java8/tree/master/datetime

Reply:
Done